packages/vue/src/components/range/MultiRange.jsx

```diff
@@ -0,0 +1,252 @@
+import { Actions, helper } from '@appbaseio/reactivecore';
+import VueTypes from 'vue-types';
+import Title from '../../styles/Title';
+import Container from '../../styles/Container';
+import { UL, Checkbox } from '../../styles/FormControlList';
+import { connect } from '../../utils/index';
+import types from '../../utils/vueTypes';
+
+const {
+  addComponent,
+  removeComponent,
+  watchComponent,
+  updateQuery,
+  setQueryListener
+} = Actions;
+
+const { isEqual, checkValueChange, getClassName } = helper;
+const MultiRange = {
+  name: 'MultiRange',
+  data() {
+
+    this.state = {
+      currentValue: [],
+      showModal: false,
+      // selectedValues hold the selected items as keys for O(1) complexity
+      selectedValues: {},
+    };
+    this.type = 'range';
```

Comment:
Why you've added these properties here?
Aren't these reactive?

Reply:
In singleRange.jsx `currentValue` is also listed in state and in the react js part as well.

Reply:
@bietkul I didn't get this what's the problem declaring `selectedValues` as state props.  do you want me to take it from `this.$data.currentValue`?

Reply:
Actually I'm talking about `this.type = 'range';` &  `this.locked = false;`, the thing is that these properties defined outside of the data ( state ) i.e not defined as reactive.
Is it intentional? since the component will not re-render when any of them will change.

Reply:
Yes, they are not defined as reactive, yes it intentional, components will not re-render when any of these will change, as state is not dependent on them, and we been have following same architecture in other vue components as well, like this https://github.com/appbaseio/reactivesearch/blob/dev/packages/vue/src/components/range/SingleRange.jsx that's why I sticked to that.

Reply:
cool